microsoft-edge/progressive-web-apps-chromium/how-to/app-actions.md

```diff
@@ -0,0 +1,402 @@
+## Prerequisites for your PWA to support app actions
+#### `share_target` in the web app manifest
+
+For a PWA to use App Actions, the `share_target` field must be present in the PWA's web app manifest file, such as [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json).
+
+* The method must be "POST".
+* The `enctype` must be `multipart/form-data`.
+* `text` and `url` in `params` are required to support app actions. 
+* `files` is optional.  You can omit `files` if you don't want your PWA to support the normal _share target_ feature.
+
+Example `share_target` declaration, from [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json):
+
+```json
+"share_target": {
+   "action": "./share-target",
+   "method": "POST",
+   "enctype": "multipart/form-data",
+   "params": {
+      "title": "title",
+      "text": "text",
+      "url": "url"
+   }
+}
+```
+
+See also:
+* [Receiving shared content](./share.md#receiving-shared-content)
+
+<!-- ====================================================================== -->
+#### `protocol_handlers` in the web app manifest
+
+For a PWA to use App Actions, the `protocol_handlers` field must be present in the PWA's web app manifest file (such as [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json)).  The `protocol_handlers` > `protocol` field must match the first part of the `invocation` > `uri` in the action definition manifest file (such as `ActionsManifest.json`).
+
+Example `protocol_handlers` declaration, from [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json):
+
+```json
+"protocol_handlers": [{
+   "protocol": "web+wami",
+   "url": "./?url=%s"
+}]
+```
+
+The matching `invocation` > `uri` field from `ActionsManifest.json`:
+
+```json
+"invocation": {
+   "type": "Uri",
+   "uri": "web+wami://paint"
+},
+```
+
+The above snippet is shown in context in [Define App Actions in the action definition manifest](#define-app-actions-in-the-action-definition-manifest), above.
+
+See also:
+* [Handle protocols in a PWA](./handle-protocols.md)
+
+
+<!-- ====================================================================== -->
+#### `launch_handler` in the web app manifest
+
+If you want to keep a single PWA window, you can set [launch_handler](https://developer.mozilla.org/docs/Web/Progressive_web_apps/Manifest/Reference/launch_handler) as `navigate-existing` in your PWA's web app manifest file (such as [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json)).
+
+Example declaration:
+
+```json
+"launch_handler": {
+   "client_mode": " navigate-existing"
+}
+```
+
+The above code is not in [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json).
+
+
+<!-- ====================================================================== -->
+#### Handle the entity of an app action
```

Comment:
This last h4 section here is not about using a web app manifest member, so it's a bit inconsistent with the previous 3 h4 sections. Also, it seems to me like what's described here is the logic that's necessary to handle `share_target` correctly. 
How about we move this to be inside the **`share_target` in the web app manifest** h4, as a new h6?